internal/controllers/device_plugin_reconciler.go

```diff
@@ -351,10 +361,12 @@ func (dsci *daemonSetCreatorImpl) setDevicePluginAsDesired(
 	standardLabels := map[string]string{constants.ModuleNameLabel: mod.Name}
 	nodeSelector := map[string]string{utils.GetKernelModuleReadyNodeLabel(mod.Namespace, mod.Name): ""}
 
-	if mod.Spec.ModuleLoader.Container.Version != "" {
+	if mod.Spec.ModuleLoader != nil && mod.Spec.ModuleLoader.Container.Version != "" {
```

Comment:
determining the labels has not become a little bit more complicated. WDYT about moving it to a dedicated function?